File: 0605-can-place-flowers/0605-can-place-flowers.cpp
class Solution {
public:
    bool canPlaceFlowers(vector<int>& flowerbed, int n) {
        int count = 0;
        int m = flowerbed.size();
        
        for(int i = 0; i < m; ++i) {
            if(flowerbed[i] == 0) {
                bool prev = (i == 0) || (flowerbed[i - 1] == 0);
                bool next = (i == m - 1) || (flowerbed[i + 1] == 0);
                
                if(prev && next) {
                    flowerbed[i] = 1;
                    count += 1;
                    
                    
                    i++;
                }
            }
            
            if (count >= n) {
                return true;
            }
        }
        
        return false;
    }

};

File: 0605-can-place-flowers/0605-can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        count = 0
        m = len(flowerbed)
        
        for i in range(m):
            if flowerbed[i] == 0:
                prev = (i == 0) or (flowerbed[i - 1] == 0)
                next = (i == m - 1) or (flowerbed[i + 1] == 0)
                
                if prev and next:
                    flowerbed[i] = 1
                    count += 1
            
            if count >= n:
                return True
        
        return False
            
        